react/fields/Dropdown/Dropdown.js

```diff
@@ -0,0 +1,239 @@
+    this.renderSelect = this.renderSelect.bind(this);
+    this.renderHelp = this.renderHelp.bind(this);
+    this.renderMessage = this.renderMessage.bind(this);
+    this.renderIcon = this.renderIcon.bind(this);
+  }
+
+  storeInputReference(input) {
+    if (input !== null) {
+      this.input = input;
+    }
+  }
+
+  renderLabel() {
+    const { label } = this.props;
+
+    if (!label) {
+      return;
+    }
+
+    const { labelProps, id } = this.props;
+    const allLabelProps = {
+      ...combineClassNames(labelProps, styles.label),
+      ...(id ? { htmlFor: id } : {})
+    };
+
+    return (
+      <label {...allLabelProps}>
+        {label}
+      </label>
+    );
+  }
+
+  renderSelect() {
+    const { inputProps = {}, id, options = [], placeholder } = this.props;
+    const inputStyles = classnames({
+      [styles.dropdown]: true,
+      [styles.blankValueSelected]: !inputProps.value
+    });
+    const allInputProps = {
+      ...combineClassNames(inputProps, inputStyles),
+      ...(id ? { id } : {}),
+      ref: this.storeInputReference,
+      value: inputProps.value || ''
+    };
+
+    return (
+      <select {...allInputProps}>
+        <option
+          className={styles.blankOption}
+          value=""
+          disabled={true}>
+          { placeholder }
+        </option>
+        {
+          options.map(({ value, label }) => (
+            <option value={value} key={value}>{ label }</option>
+          ))
+        }
+      </select>
+    );
+  }
+
+  renderChevron() {
+    return (
+      <ChevronIcon
+        className={styles.chevron}
+        svgClassName={styles.chevronSvg}
+        direction="down"
+      />
+    );
+  }
+
+  renderHelp() {
+    const { message, help } = this.props;
+
+    if (message || !help) {
+      return;
+    }
+
+    const { helpProps } = this.props;
+    const allHelpProps = combineClassNames(helpProps, styles.help);
+
+    return (
+      <p {...allHelpProps}>
+        {help}
+      </p>
+    );
+  }
+
+  renderMessage() {
+    const { message } = this.props;
+
+    if (!message) {
+      return;
+    }
+
+    const { messageProps } = this.props;
+    const allMessageProps = combineClassNames(messageProps, styles.message);
+
+    return (
+      <p {...allMessageProps}>
+        {this.renderIcon()}
+        {message}
+      </p>
+    );
+  }
+
+  renderIcon() {
+    const { invalid } = this.props;
+
+    if (!invalid) {
+      return;
+    }
+
+    return (
+      <ErrorIcon
+        filled={true}
+        className={styles.messageIcon}
+        svgClassName={styles.messageIconSvg}
+      />
+    );
+  }
+
+  render() {
+    const { className, invalid } = this.props;
+    const classNames = classnames({
+      [styles.root]: true,
+      [styles.invalid]: invalid,
+      [className]: className
+    });
+
+    return (
+      <div className={classNames}>
+        {this.renderLabel()}
+        {this.renderChevron()}
+        {this.renderSelect()}
```

Comment:
Everything in the style guide should be compatible in my mind. I would encourage to check this before merging.